Applications/Spire/Source/Ui/EditableTableView.cpp

```diff
@@ -282,12 +320,66 @@ QWidget* EditableTableView::make_table_item(const ViewBuilder& view_builder,
   } else if(column == table->get_column_size() - 1) {
     return make_empty_cell();
   } else {
-    return view_builder(
+    auto item = view_builder(
       std::static_pointer_cast<EditableTableModel>(get_table())->m_source,
       any_cast<int>(table->at(row, 0)), column - 1);
+    if(auto editable_box = find_editable_box(*item)) {
```

Comment:
This is too fragile, why does the EditableTableView make any assumption about the returned widget being an EditableBox? If an EditableBox is needed then that needs to be encoded in the type system. Try to get comfortable with using the type system and API design to encode your requirements and communicate what is expected of the EditableTableView.